src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            FruitTransaction.Operation operation = getOperationByCode(parts[0]);
+            String fruit = parts[1];
+            int quantity = Integer.parseInt(parts[2]);
+            transactions.add(new FruitTransaction(operation, fruit, quantity));
+        }
+        return transactions;
```

Comment:
The method 'convertToTransaction' seems to handle multiple responsibilities: it parses the data, skips headers, and creates transactions. Consider refactoring this method to follow the Single Responsibility Principle (one of the SOLID principles). You could, for instance, extract the header skipping logic and the parsing logic into separate methods or classes.